ToolBox/CoinApiDataConverter/CoinApiDataReader.cs

```diff
@@ -79,48 +66,79 @@ public CoinApiEntryData GetCoinApiEntryData(TarInputStream tar, TarEntry entry)
         /// <param name="tar">The tar input stream</param>
         /// <param name="entryData">The entry data</param>
         /// <returns>An <see cref="IEnumerable{Tick}"/> for the ticks read from the entry</returns>
-        public IEnumerable<Tick> ProcessCoinApiEntry(TarInputStream tar, CoinApiEntryData entryData)
+        public IEnumerable<Tick> ProcessCoinApiEntry(CoinApiEntryData entryData, FileInfo file)
         {
             Log.Trace("CoinApiDataReader.ProcessTarEntry(): Processing " +
                       $"{entryData.Symbol.ID.Market}-{entryData.Symbol.Value}-{entryData.TickType} " +
                       $"for {entryData.Date:yyyy-MM-dd}");
 
-            using (var gzipStream = new MemoryStream())
+
+            var tickList = new List<Tick>();
+
+            var csv = ExtractFile(file);
+
+            using (var innerStream = csv.OpenRead())
+            using (var reader = new StreamReader(innerStream))
+            {
+                var headerLine = reader.ReadLine();
+                if (headerLine == null)
+                {
+                    throw new Exception($"CoinApiDataReader.ProcessCoinApiEntry(): CSV header not found for entry name: {entryData.Name}");
+                }
+
+                var headerParts = headerLine.Split(';').ToList();
+
+                tickList = entryData.TickType == TickType.Trade
+                    ? ParseTradeData(entryData.Symbol, reader, headerParts)
+                    : ParseQuoteData(entryData.Symbol, reader, headerParts);
+            }
+
+            csv.Delete();
+
+            return tickList;
+        }
+
+        private FileInfo ExtractFile(FileInfo source)
+        {
+            var zipper = OS.IsWindows ? "C:/Program Files/7-Zip/7z.exe" : "7z";
```

Comment:
Should this be a config or command line option?
I think international versions of Windows have the `Program Files` folder name localized.

Reply:
Is there a specific reason for using 7Zip and extracting to disk vs. unzipping to memory as before?
As far as I understand, with the addition of parallelism, we would never be reading from the same file from multiple threads, right?

Reply:
Not reallly, reverting and using `System.IO.Compression` GZip implementation.